bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/internal/JdbcConfiguration.java

```diff
@@ -40,12 +43,12 @@ public class JdbcConfiguration {
 
     private Map<Object, Object> configuration;
 
-    private JdbcBaseDAO dBDAO = null;
-    private String dbName = null;
+    private @NonNullByDefault({}) JdbcBaseDAO dBDAO;
```

Comment:
I have the feeling `dBDAO` can be null when `updateConfig` returns false before setting it.

Reply:
This was a compromise which reflects the current situation/risk. I did some analysis before leaving it like this:
- Before `getDBDAO` can be trusted to return non-null, `setDBDAOClass` must be called and return true.
- `setDBDAOClass` is called by `updateConfig`.
- `updateConfig` is called by constructor.
- Only after successful validation will `initialized` be set to true in JdbcPersistenceService (which extends JdbcMapper).
- It seems that the methods that will end up calling `getDBDAO` will first check `initialized` directly or indirectly through `checkDBAccessability`. There are many paths to check though, and I'm not completely sure I didn't miss any.

To be honest, this is kind of a mess. The commented change should not change anything runtime - any `NullPointerException` thrown now should also be thrown before this change. In order to make this `@NonNull`, I believe a lot of refactoring would be needed.

Any thoughts/ideas?

Reply:
Let's keep it like that if you are confident.